src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,33 @@
+import { useParams } from "react-router-dom";
+import { useLocation } from "react-router-dom";
+
+function SingleArticle() {
+  const { article_id } = useParams();
+  const location = useLocation();
+  const articles = location.state;
```

Comment:
using state from location here is creating big issues if I go straight to /articles/3 for example. This is another reason to use a fresh API call for single article.